verilog/CST/tasks_test.cc

```diff
@@ -222,5 +225,49 @@ TEST(GetTaskIdTest, QualifiedIds) {
   }
 }
 
+TEST(GetTaskHeaderTest, GetTaskName) {
+  constexpr int kTag = 1;  // value doesn't matter
+  const SyntaxTreeSearchTestCase kTestCases[] = {
+      {""},
+      {"module m(); endmodule: m"},
+      {"task ", {kTag, "foo"}, "(int a, bit b);\nendtask"},
+      {"task ",
+       {kTag, "foo"},
+       "();\n endtask\n task ",
+       {kTag, "bar"},
+       "()\n; endtask"},
+      {"module my_module;\ntask ",
+       {kTag, "inner_task"},
+       "(int my_args);\n$display(my_arg2);\nendtask\nendmodule"},
+      {"class task_class;\ntask ",
+       {kTag, "my_task"},
+       "(input int a, b, output int c);\nc = a + b;\nendtask\nendclass"},
+      {"package my_pkg;\ntask automatic ",
```

Comment:
Also:

* `virtual` task inside class)
* `static` task (inside and outside class)

Reply:
@fangism  Is it possible to write a static task or function outside classes ?
I tried that but got syntax errors.

Reply:
The syntax should be `function static ...` or `function automatic ...`